generators/dotnet-templates/Microsoft.BotFramework.CSharp.CoreBot/content/.template.config/template.json

```diff
@@ -45,16 +45,12 @@
         "datatype": "choice",
         "choices": [
           {
-            "choice": "netcoreapp2.2",
-            "description": "Target netcoreapp2.2"
-          },
-          {
-            "choice": "netcoreapp2.1",
-            "description": "Target netcoreapp2.1"
+            "choice": "netcoreapp3.1",
```

Comment:
I'm confused about what our plan is regarding the various supported versions of the .NET Core.  Based on the commit here, we *only* support 3.1.  This doesn't seem correct.  It seems we should support all supported .NET Core versions.  this would imply 2.2, no?

Reply:
Our libs support both 2.2 and 3.1, as per the plan of record.  Our sample though, that consume the libs, need to target a version of .NET Core. The direction I was given from @cleemullins is that we should shift our samples to 3.1  We have also created a branch of the samples and marked it at 2.2 so that folks have a branch of those samples.

I think our only other option would be to either manage two copies of our samples which feels unmanageable or maybe we could add a couple of specific .NET 2.2 targeted samples such as for Core Bot?

Thoughts @sgellock?

Reply:
Sorry - I am clearly tired, hence I deleted my last comment which was talking about samples.

I think we have two choices - you are correct that we are supporting both 2.2 and 3.1.

1.  We have our templates target both 2.2 and 3.1 but this will cause complexity as we will need to have conditionals through the template in places.

2. Our SDK still supports both, but our templates target 3.1.

Reply:
FWIW, I think supporting 2.2 is a good thing and necessary with the libs, but we don't have to continue to have our templates target both from my perspective.

Reply:
I have spoken with customers that have  use --framework to set the runtime version they are using.  this needs to be a design goal that we maintain.

Reply:
Understood - leave it with me.

Reply:
RE:  complexity of the templates - yes.  this will create complexity in our code but that complexity provides flexibility for our customers.  I'm willing to make this trade off